internal/cli/up.go

```diff
@@ -247,3 +264,38 @@ Usage: waypoint up [options]
 
 ` + c.Flags().Help())
 }
+
+// Helper functions for formatting variable final value output
+func fmtVariablesOutput(values map[string]*pb.Variable_FinalValue) *terminal.Table {
+	headers := []string{
+		"Variable", "Value", "Type", "Source",
+	}
+	tbl := terminal.NewTable(headers...)
+	output := formatter.ValuesForOutput(values)
+	var columns []string
+	for iv, v := range output {
+		// We add a line break in the value here because the Table word wrap
+		// alone can't accomodate the column headers to a long value
+		if len(v.Value) > 45 {
+			for i := range v.Value {
+				// line break every 45 characters
+				if i%46 == 0 && i != 0 {
+					v.Value = v.Value[:i] + "\n" + v.Value[i:]
+				}
+			}
+		}
```

Comment:
[See my other comment](https://github.com/hashicorp/waypoint/pull/3138#issuecomment-1079226152)

This code crops the first chunk as 46 chars, and every subsequent chunk as 45 chars. This code below should work to crop every chunk as 45 chars:

```suggestion
	if len(v.Value) > 45 {
		for i := 45; i < len(v.Value); i += 46 {
			v.Value = v.Value[:i] + "\n" + v.Value[i:]
		}
	}
```